### PR TITLE
♻️ do not send heartbeat as soon as the service is started

### PR DESCRIPTION
**Pull request by GitHK**
## What do these changes do?

The heart beat was sent as soon as the service was started. Since the order of events is not guaranteed, it can arrive before the start event.

With @matusdrobuliak66 we have decided to skip the first heartbeat event.

Bonus:
- fixes flaky tests in `tests/unit/test_modules_dynamic_sidecar_client_api_base.py`: `test_connection_error` and possibly flaky `test_retry_on_errors_by_error_type` 

## Related issue/s

- https://github.com/ITISFoundation/osparc-simcore/issues/5169

## Dev Checklist

- [x] No ENV changes or I properly updated ENV ([read the instruction](https://git.speag.com/oSparc/osparc-ops-deployment-configuration/-/blob/configs/README.md?ref_type=heads#how-to-update-env-variables))

### CODE REVIEW

**Review by sanderegg — CHANGES_REQUESTED** [review]
I see the point, but maybe have a look at my suggestion. I find very strange to change the private members of a module to then call them from outside

**Review by matusdrobuliak66 — APPROVED** [review]
thanks!

File: packages/service-library/src/servicelib/decorators.py [hunk 1]
```diff
@@ -20,16 +24,29 @@ def safe_return(if_fails_return=False, catch=None, logger=None):
...
+            return deepcopy(if_fails_return)  # avoid issues with default mutable
 
         return safe_func
 
     return decorate
+
+
+def async_delayed(
+    interval: datetime.timedelta,
+) -> Callable[..., Callable[..., Coroutine]]:
```
Comment (sanderegg) [inline packages/service-library/src/servicelib/decorators.py:43]: Q: must this be a Coroutine? not Awaitable[Any] ?

**Review by sanderegg — APPROVED** [review]
thanks for the changes

File: packages/service-library/src/servicelib/decorators.py (hunk 1 above)
Comment (GitHK) [reply packages/service-library/src/servicelib/decorators.py:43]: Apparently this plays nice with `asyncio.create_task(` which expects a coroutine. also it is correct, `wrapper` is a coroutine. So I think it's OK like this.

File: packages/service-library/src/servicelib/background_task.py [hunk 2]
```diff
@@ -48,18 +50,22 @@ def start_periodic_task(
...
             _periodic_scheduled_task(
                 task,
                 interval=interval,
                 task_name=task_name,
                 **kwargs,
-            ),
+            )
+        )
+        return asyncio.create_task(
+            delayed_periodic_scheduled_task(),
```
[the remark this answers is not in the document]
Comment (GitHK) [reply packages/service-library/src/servicelib/background_task.py:68]: I also thought the same thing. This is correct. If you think about it (it took me a while), if you decorate a function, you still call it where you use it. This is the same.

**Pull request merged**